File: cai/83.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head == None:
            return head
        now = head
        nex = head.next
        while nex != None:
           if now.val==nex.val:
               now.next=nex.next
               nex=now.next
           else:
               now=now.next
               nex=nex.next
        return head

File: xiong/83.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head==NULL) return head;
        if(head->next == NULL) return head;
        ListNode *first = head, *second = head->next;
        if(first == NULL || second == NULL) return head;
        while(second != NULL){
            if(first->val == second->val){
                first->next = second->next;
                second = first->next;
            }else{
                first = first->next;
                second = second->next;
            }
        }
        // ListNode *t = head;
        // while(t->next != NULL){
        //     cout<<(t->val);
        //     t = t->next;
        // }
        return head;
        
    }
};